tracer/build/_build/Build.Steps.cs

```diff
@@ -1053,6 +1040,74 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
                 .SetMaxCpuCount(null));
         });
 
+    Target CompileDebuggerIntegrationTests => _ => _
+        .Unlisted()
+        .After(CompileManagedSrc)
+        .DependsOn(CompileDebuggerIntegrationTestsDependencies)
+        .DependsOn(CompileDebuggerIntegrationTestsSamples)
+        .Requires(() => Framework)
+        .Requires(() => MonitoringHomeDirectory != null)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetConfiguration(BuildConfiguration)
+                .SetFramework(Framework)
+                .SetTargetPlatformAnyCPU()
+                .EnableNoRestore()
+                .EnableNoDependencies()
+                .SetNoWarnDotNetCore3()
+                .SetProjectFile(DebuggerIntegrationTests));
+        });
+
+    Target CompileDebuggerIntegrationTestsDependencies => _ => _
+        .Unlisted()
+        .Requires(() => Framework)
+        .Requires(() => MonitoringHomeDirectory != null)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetConfiguration(BuildConfiguration)
+                .SetFramework(Framework)
+                .SetTargetPlatformAnyCPU()
+                .EnableNoRestore()
+                .SetNoWarnDotNetCore3()
+                .SetProjectFile(DebuggerSamplesTestRuns));            
+            
+            DotNetBuild(s => s
+                .SetConfiguration(BuildConfiguration)
+                .SetFramework(Framework)
+                .SetTargetPlatform(TargetPlatform)
+                .EnableNoRestore()
+                .SetNoWarnDotNetCore3()
+                .SetProjectFile(DebuggerSamplesTestRuns));
+        });
+
+    Target CompileDebuggerIntegrationTestsSamples => _ => _
+        .Unlisted()
+        .DependsOn(CompileDebuggerIntegrationTestsDependencies)
+        .Requires(() => Framework)
+        .Requires(() => MonitoringHomeDirectory != null)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetFramework(Framework)
+                .SetConfiguration(BuildConfiguration)
+                .SetTargetPlatform(TargetPlatform)
+                .EnableNoRestore()
+                .EnableNoDependencies()
+                .SetNoWarnDotNetCore3()
+                .SetProjectFile(DebuggerSamples));
+
+            if (!IsWin)
+            {
+                DotNetPublish(x => x
```

Comment:
nit: might be nice to explain why given it took us a while to figure out 😄 
```suggestion
                // The sample helper in the test library assumes that the sample has 
                // been published when running on Linux
                DotNetPublish(x => x
```